h/static/scripts/directive/excerpt.coffee

```diff
@@ -0,0 +1,48 @@
+###*
+# @ngdoc directive
+# @name excerpt
+# @restrict AE
+# @description Checks to see if text is overflowing its container.
+# If so, it prepends/appends expanders/collapsers. Note, to work
+# the element needs a max height.
+###
+module.exports = ->
+  link: (scope, elem, attr, ctrl) ->
+    scope.collapsed = true
+
+    scope.isOverflowing = ->
+      excerpt = elem[0].querySelector('.excerpt')
+      if excerpt == null
+        return false
+      return elem[0].scrollHeight > elem[0].clientHeight
+
+    scope.toggle = ->
+      scope.collapsed = !scope.collapsed
+
+  scope:
+    enabled: '=excerptIf'
+    maxheight: '=maxheight'
+    bottomGradient: '=excerptBottomGradient'
+  restrict: 'AE'
+  transclude: true
+  template: '''
+    <ng-transclude></ng-transclude>
+
+    <div class="excerpt-wrapper">
```

Comment:
The reason I took it out was it's a bit to general for our use case, especially since we are transcluding the markdown editor as well:
![hypothesis](https://cloud.githubusercontent.com/assets/521978/10028086/5822b6c6-611f-11e5-9bb7-aa2e4615ec92.png)

In the latest version I've also added an optional `excerpt-show-controls` attribute, which I use on the annotation body to prevent this problem.